server/src/main/java/org/elasticsearch/bootstrap/EntitlementBootstrap.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.bootstrap;
+
+import com.sun.tools.attach.AgentInitializationException;
+import com.sun.tools.attach.AgentLoadException;
+import com.sun.tools.attach.AttachNotSupportedException;
+import com.sun.tools.attach.VirtualMachine;
+
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+
+import java.io.IOException;
+
+final class EntitlementBootstrap {
+
+    static void configure() {
+        logger.debug("Loading agent");
+        try {
+            VirtualMachine vm = VirtualMachine.attach(Long.toString(ProcessHandle.current().pid()));
+            try {
+                vm.loadAgent(agentJar());
+            } finally {
+                vm.detach();
+            }
+        } catch (AttachNotSupportedException | IOException | AgentLoadException | AgentInitializationException e) {
+            throw new IllegalStateException("Unable to attach entitlement agent", e);
+        }
+    }
+
+    private static String agentJar() {
+        String result = System.getProperty("es.entitlement.agentJar");
+        if (result == null) {
+            throw new IllegalStateException("Must set es.entitlement.agentJar property");
+        }
+        return result;
+    }
+
+    private static final Logger logger = LogManager.getLogger(EntitlementBootstrap.class);
```

Comment:
Heh I keep forgetting.

Reply:
To explain myself: I tend to favour "use before declaration" ordering because that makes it easier to read code from beginning to end. (It happens to be the opposite of the order mandated by older languages like C or Pascal, which were designed to enable compilation in a single pass over the source code. In other words, they favoured compilability over readability.) I also favour grouping all the instance fields of a class together at the top so you can tell at a glance what the object is composed of, which tends to help quickly give the reader the gist of what the object does, and is pretty easy to skip visually if you actually don't care. The logger declaration is boilerplate that does nothing to illuminate how the class works, so it gets the lowest priority and end up at the bottom.

But of course, all of this is outweighed by the benefits of uniformity, so I ought to be following our conventions. 😅